src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "OTAMultiImageProcessorImpl.h"
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+// TODO: Add this dependency when integrating the multi-OTA encryption feature
+// #include <platform/silabs/SilabsConfig.h>
+
+extern "C" {
+#include "sl_si91x_driver.h"
+}
+
+namespace chip {
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessInternal(ByteSpan & block)
+{
+    sl_status_t status = SL_STATUS_OK;
+
+    ChipLogProgress(SoftwareUpdate, "Process WiFi Block");
+    // The first block is expected to be the OTA descriptor.
+    // `ProcessDescriptor` handles it and sets `mDescriptorProcessed` to avoid reprocessing.
+    if (!mDescriptorProcessed)
+    {
+        ReturnErrorOnFailure(ProcessDescriptor(block));
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+        /* 16 bytes to used to store undecrypted data because of unalignment */
+        mAccumulator.Init(requestedOtaMaxBlockSize + 16);
+#endif // SL_MATTER_ENABLE_OTA_ENCRYPTION
+    }
+
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+    MutableByteSpan mBlock = MutableByteSpan(mAccumulator.GetData(), mAccumulator.GetThreshold());
+    memcpy(&mBlock[0], &mBlock[requestedOtaMaxBlockSize], mUnalignmentNum);
+    memcpy(&mBlock[mUnalignmentNum], block.data(), block.size());
+
+    if (mUnalignmentNum + block.size() < requestedOtaMaxBlockSize)
+    {
+        uint32_t mAlignmentNum = (mUnalignmentNum + block.size()) / 16;
+        mAlignmentNum          = mAlignmentNum * 16;
+        mUnalignmentNum        = (mUnalignmentNum + block.size()) % 16;
+        memcpy(&mBlock[requestedOtaMaxBlockSize], &mBlock[mAlignmentNum], mUnalignmentNum);
+        mBlock.reduce_size(mAlignmentNum);
+    }
+    else
+    {
+        mUnalignmentNum = mUnalignmentNum + block.size() - requestedOtaMaxBlockSize;
+        mBlock.reduce_size(requestedOtaMaxBlockSize);
+    }
+
+    OTATlvProcessor::vOtaProcessInternalEncryption(mBlock);
+    block = mBlock;
```

Comment:
The input block is passed by reference so we are modyfing the input block here, is that intended?

Reply:
For reference, I have added these for encryption. This may be modified as part of enabling multi-ota encryption PR changes.